packages/explat-client-react-helpers/src/index.tsx

```diff
@@ -53,17 +79,17 @@ export default function createExPlatClientReactHelpers(
 
 		useEffect( () => {
 			let isSubscribed = true;
-			exPlatClient.loadExperimentAssignment( experimentName ).then( ( experimentAssignment ) => {
-				if ( isSubscribed ) {
-					setState( [ false, experimentAssignment ] );
-				}
-			} );
+			if ( options.isEligible ) {
+				exPlatClient.loadExperimentAssignment( experimentName ).then( ( experimentAssignment ) => {
```

Comment:
You might want to add
```js
setState( [ true, null ] );
```
above this line, before `loadExperimentAssignment` starts. Then when the `isEligible` is toggled from `false` to `true`, the progression of states is:
1. `[ false, null ]` -- not eligible, not loading, the `null` assignment
2. `[ true, null ]` -- is eligible and is loading, assignment still `null`
3. `[ false, final ]` -- is eligible and finished loading, assignment is `final`.

Without the extra `setState`, the `isLoading` state remains `false` even though a load is in progress.

Reply:
Very good catch there!

Reply:
Ahh, actually this isn't necessary since we never set the state to `[ false, null ]`, we even test for it here:
https://github.com/Automattic/wp-calypso/pull/51565/files#diff-910569eaf3e8b5a6129b7c2f3102c9569aff72f81f782584981cc5bffe5bd487R108

So the eligible state is kept separate from the not-eligible.

Reply:
I see, I missed that `state` is ever used (set or read) only if `isEligible` is `true` 🙂